control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go

```diff
@@ -609,3 +654,49 @@ func inPlaceUpgradeConfigMap(poolName string, namespace string) *corev1.ConfigMa
 		},
 	}
 }
+
+func inPlaceUpgradeDaemonServiceAccount(namespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "machine-config-daemon",
+		},
+	}
+}
+
+func inPlaceUpgradeDaemonClusterRole(namespace string) *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("machine-config-daemon-%s", namespace),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{"security.openshift.io"},
```

Comment:
can you elaborate why is this needed now?

Reply:
I assume this is for a good reason which we should not bypass?

Reply:
This rule matches the self-driving OCP's MCD: https://github.com/openshift/machine-config-operator/blob/b6cfd42372427eb5d7aee44ce30c81c74ae12b08/manifests/machineconfigdaemon/clusterrole.yaml#L12-L15

The only way I am aware of bypassing it is to use `run-level: 1` which we removed from the MCO: https://github.com/openshift/machine-config-operator/commit/8c4dda1cc54bc85f951cd3a78bb7ba8a55a9f8bf

That seems even worse in terms of security, and I don't know of another way to do so